### ISSUE TITLE
Spring Cloud Spanner - Support "Set"  Type

### ISSUE THREAD

**Issue by ps882** [GoogleCloudPlatform/spring-cloud-gcp#3291]
We'd love to accept your patches and contributions to this project. There are just a few small
guidelines you need to follow before opening an issue or a PR:

1. Ensure the issue was not already reported.
2. Open a new issue if you are unable to find an existing issue addressing your problem. Make sure
   to include a title and clear description, as much relevant information as possible, and a code
   sample or an executable test case demonstrating the expected behavior that is not occurring.
3. Discuss the priority and potential solutions with the maintainers in the issue. The maintainers
   would review the issue and add a label "Accepting Contributions" once the issue is ready for
   accepting contributions.
4. Open a PR only if the issue is labeled with "Accepting Contributions", ensure the PR description
   clearly describes the problem and solution. Note that an open PR without an issue labeled with "
   Accepting Contributions" will not be accepted.

See
also [CONTRIBUTING.md](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/blob/main/CONTRIBUTING.md)
.

**Is your feature request related to a problem? Please describe.**
I have a requirement where i need to store distinct values (labels) in my table but currently Spanner only allows "List" and not "Set". Which means i have to do the validation that distinct items are stored, on my own and then save the values in SpannerRepository.

An interesting observation was, when i define the field as "Set" and then call save() it gets saved. But when i try to read it gives me casting error that it cannot cast from List to Set. Which means SpannerRepository takes the Set field and saves it as List because that's what it supports

**Describe the solution you'd like**
Spanner Spring supports Set type so that i can define my pojo label field as "Set"

**Describe alternatives you've considered**
Ensuring the labels are unique on my own.

**Additional context**

https://googlecloudplatform.github.io/spring-cloud-gcp/3.2.1/reference/html/spanner.html#_supported_types

Thanks!
